tests/test_sca.py

```diff
@@ -38,60 +37,61 @@ def test_parse_components():
         "/rpms/nodejs",
         "3cbed2be4171502499d0d89bea1ead91690af7d2",
         "nodejs",
-        "rpms",
-        "tests/data/rpms/nodejs",
+        1,
+        "/tmp/1",
         True,
     ),
     (
         f"git://{os.getenv('CORGI_LOOKASIDE_CACHE_URL')}"  # Comma not missing, joined with below
         "/containers/openshift-enterprise-console",
         "f95972ce68d2850ae20c10fbf87182a17fa24b19",
         "openshift-enterprise-console",
-        "containers",
-        "tests/data/containers/openshift-enterprise-console",
+        2,
+        "/tmp/2",
         False,
     ),
 ]
 
 
 @pytest.mark.parametrize(
-    "source_url,commit,package_name,package_type,expected_path,path_exists",
+    "source_url,commit,package_name,build_id,expected_path,path_exists",
     archive_source_test_data,
 )
 @patch("subprocess.check_call")
 def test_clone_source(
-    mock_subprocess, source_url, commit, package_name, package_type, expected_path, path_exists
+    mock_subprocess, source_url, commit, package_name, build_id, expected_path, path_exists
 ):
     # This prevents multiple runs of this test from having different results because
     # we mkdir the directory prior to the clone
     if not path_exists:
         shutil.rmtree(expected_path, ignore_errors=True)
-        _clone_source(f"{source_url}#{commit}", package_type)
-        print(mock_subprocess.call_arg_list)
+        _clone_source(f"{source_url}#{commit}", build_id)
         mock_subprocess.assert_has_calls(
             [
                 call(["/usr/bin/git", "clone", source_url, PosixPath(expected_path)]),
-                call(["/usr/bin/git", "checkout", commit], cwd=PosixPath(expected_path)),
+                call(["/usr/bin/git", "checkout", commit], cwd=PosixPath(expected_path), stderr=-3),
             ]
         )
     else:
+        expected_path_obj = Path(expected_path)
+        if not expected_path_obj.exists():
+            os.mkdir(f"/tmp/{build_id}")
         with pytest.raises(FileExistsError):
-            _clone_source(f"{source_url}#{commit}", package_type)
+            _clone_source(f"{source_url}#{commit}", build_id)
         mock_subprocess.assert_not_called()
 
 
 @patch("subprocess.check_call")
 @patch("corgi.tasks.sca._download_lookaside_sources")
 def test_get_distgit_sources(mock_check_call, mock_download_lookaside_sources):
-    expected_path = "tests/data/rpms/test"
+    shutil.rmtree("/tmp/1")
+    expected_path = "/tmp/1"
```

Comment:
Minor nit: Instead of hardcoding these, could we move them into a variable like test_build_id = 1?